docs/GeoReplication.md

```diff
@@ -0,0 +1,137 @@
+
+## Configuring replication
+
+### Granting permissions to properties
+
+When each tenant is on-boarded to a Pulsar cluster, it needs to be
+granted permissions to be able to create namespaces in any particular cluster.
+
+When creating a new property, specify all the intended clusters:
+```shell
+$ bin/pulsar-admin properties create my-property    \
+        --admin-roles my-admin-role                 \
+        --allowed-clusters us-west,us-east,us-cent
+```
+
+You can update the configuration of an existing property by using `update`
+instead of `create`.
+
+### Creating the namespace
+
+To use replication, we need a *global* topic, that is to say, a topic
+that belong to a *global* namespace, not tied to any particular cluster.
+
+Global namespaces are assigned to the `global` virtual cluster. To create a new
+namespace:
+
+```shell
+$ bin/pulsar admin namespaces create my-property/global/my-namespace
+```
+
+Initially, the namespace is not assigned to any cluster.
+
+```shell
+$ bin/pulsar admin namespaces set-clusters
+                          my-property/global/my-namespace \
+                          --clusters us-west,us-east,us-cent
+```
+
+The list of replication clusters can be changed at any time, with no
+disruption to ongoing traffic. Replication channels will be immediately
+set up or stopped in all the clusters as soon as the configuration changes.
+
+### Using global topics
+
+At this point you can just create a producer or consumers and automatically
+create a global topic.
+
+Typically, each application will use the `serviceUrl` for the local cluster.
+
+#### Selective replication
+
+It is possible to override the per-namespace list of replication clusters for
+each individual message by restricting the replication list.
+
+In the Java API, this is done on the message builder:
+```java
+...
+List<String> restrictReplicationTo = new ArrayList<>();
+restrictReplicationTo.add("us-west");
+restrictReplicationTo.add("us-east");
+
+Message message = MessageBuilder.create()
+              .setContent("my-payload".getBytes())
+              .setReplicationClusters(restrictReplicationTo)
+              .build();
+
+producer.send(message);
+```
+
+#### Topic stats
+
+Topic stats for global topics are available through the same API and tools:
+
+```shell
+$ bin/pulsar-admin persistent stats persistent://my-property/global/my-namespace/my-topic
+```
+
+Each cluster will report its own local stats, including incoming and outgoing
+replication rates and backlogs.
+
+#### Deleting a global topic
+
+Given that the topic exist in multiple regions, it's not possible to directly
+delete a global topic. The procedure is to rely on the automatic topic garbage
+collection.
+
+In Pulsar, a topic is automatically deleted when it's not used anymore, that is
+to say, when no producers or consumers are connected ***and*** there are no
+subscriptions.
+
+For global topics, each region will use a fault-tolerant mechanism to decide
```

Comment:
To delete a global topic, close all producers and consumers on the topic and delete all its local subscriptions in every replication cluster.  When Pulsar determines that no valid subscription for the topic remains  across the system, it will garbage collect the topic.